File: Nx3CS/bin/test/src/nx3/PBeamgroupDirectionCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PBeamgroupDirectionCalculator : global::haxe.lang.HxObject {
		public    PBeamgroupDirectionCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PBeamgroupDirectionCalculator(global::nx3.PBeamgroup beamgroup){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				global::nx3.PBeamgroupDirectionCalculator.__hx_ctor_nx3_PBeamgroupDirectionCalculator(this, beamgroup);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PBeamgroupDirectionCalculator(global::nx3.PBeamgroupDirectionCalculator __temp_me87, global::nx3.PBeamgroup beamgroup){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				__temp_me87.beamgroup = beamgroup;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				return new global::nx3.PBeamgroupDirectionCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				return new global::nx3.PBeamgroupDirectionCalculator(((global::nx3.PBeamgroup) (arr[0]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PBeamgroup beamgroup;
		
		public virtual   global::nx3.EDirectionUD getDirection(){
			unchecked {
				#line 18 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				this.topLevel = this.findTopLevel();
				this.bottomLevel = this.findBottomLevel();
				if (( ( this.topLevel + this.bottomLevel ) <= 0 )) {
					#line 20 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					return global::nx3.EDirectionUD.Down;
				}
				
				#line 21 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				return global::nx3.EDirectionUD.Up;
			}
			#line default
		}
		
		
		public  int topLevel;
		
		public virtual   int findTopLevel(){
			unchecked {
				#line 28 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				int topLevel = ((global::nx3.PNote) (this.beamgroup.pnotes[0]) ).nnote.getTopLevel();
				if (( this.beamgroup.pnotes.length == 1 )) {
					#line 29 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					return topLevel;
				}
				
				#line 30 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				{
					#line 30 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					int _g1 = 1;
					#line 30 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					int _g = this.beamgroup.pnotes.length;
					#line 30 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					while (( _g1 < _g )){
						#line 30 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						int i = _g1++;
						#line 32 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						int level = ((global::nx3.PNote) (this.beamgroup.pnotes[i]) ).nnote.getTopLevel();
						{
							#line 33 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
							double x = global::System.Math.Min(((double) (topLevel) ), ((double) (level) ));
							#line 33 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
							topLevel = ((int) (x) );
						}
						
					}
					
				}
				
				#line 35 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				return topLevel;
			}
			#line default
		}
		
		
		public  int bottomLevel;
		
		public virtual   int findBottomLevel(){
			unchecked {
				#line 41 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				int bottomLevel = ((global::nx3.PNote) (this.beamgroup.pnotes[0]) ).nnote.getBottomLevel();
				if (( this.beamgroup.pnotes.length == 1 )) {
					#line 42 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					return bottomLevel;
				}
				
				#line 43 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				{
					#line 43 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					int _g1 = 1;
					#line 43 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					int _g = this.beamgroup.pnotes.length;
					#line 43 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					while (( _g1 < _g )){
						#line 43 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						int i = _g1++;
						#line 45 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						int level = ((global::nx3.PNote) (this.beamgroup.pnotes[i]) ).nnote.getBottomLevel();
						{
							#line 46 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
							double x = global::System.Math.Max(((double) (bottomLevel) ), ((double) (level) ));
							#line 46 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
							bottomLevel = ((int) (x) );
						}
						
					}
					
				}
				
				#line 48 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				return bottomLevel;
			}
			#line default
		}
		
		
		public override   double __hx_setField_f(string field, int hash, double @value, bool handleProperties){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				switch (hash){
					case 1475586713:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						this.bottomLevel = ((int) (@value) );
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return @value;
					}
					
					
					case 1077093039:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						this.topLevel = ((int) (@value) );
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return base.__hx_setField_f(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				switch (hash){
					case 1475586713:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						this.bottomLevel = ((int) (global::haxe.lang.Runtime.toInt(@value)) );
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return @value;
					}
					
					
					case 1077093039:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						this.topLevel = ((int) (global::haxe.lang.Runtime.toInt(@value)) );
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return @value;
					}
					
					
					case 1104035696:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						this.beamgroup = ((global::nx3.PBeamgroup) (@value) );
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				switch (hash){
					case 419158752:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("findBottomLevel") ), ((int) (419158752) ))) );
					}
					
					
					case 1475586713:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return this.bottomLevel;
					}
					
					
					case 766578696:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("findTopLevel") ), ((int) (766578696) ))) );
					}
					
					
					case 1077093039:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return this.topLevel;
					}
					
					
					case 529989321:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getDirection") ), ((int) (529989321) ))) );
					}
					
					
					case 1104035696:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return this.beamgroup;
					}
					
					
					default:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_getField_f(string field, int hash, bool throwErrors, bool handleProperties){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				switch (hash){
					case 1475586713:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return ((double) (this.bottomLevel) );
					}
					
					
					case 1077093039:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return ((double) (this.topLevel) );
					}
					
					
					default:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return base.__hx_getField_f(field, hash, throwErrors, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				switch (hash){
					case 419158752:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return this.findBottomLevel();
					}
					
					
					case 766578696:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return this.findTopLevel();
					}
					
					
					case 529989321:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return this.getDirection();
					}
					
					
					default:
					{
						#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				baseArr.push("bottomLevel");
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				baseArr.push("topLevel");
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				baseArr.push("beamgroup");
				#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
				{
					#line 8 "F:\\nx3\\src\\nx3\\PBeamgroupDirectionCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/PBeamgroupDirectionCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PBeamgroupDirectionCalculator extends haxe.lang.HxObject
{
	public    PBeamgroupDirectionCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PBeamgroupDirectionCalculator(nx3.PBeamgroup beamgroup)
	{
		nx3.PBeamgroupDirectionCalculator.__hx_ctor_nx3_PBeamgroupDirectionCalculator(this, beamgroup);
	}
	
	
	public static   void __hx_ctor_nx3_PBeamgroupDirectionCalculator(nx3.PBeamgroupDirectionCalculator __temp_me92, nx3.PBeamgroup beamgroup)
	{
		__temp_me92.beamgroup = beamgroup;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PBeamgroupDirectionCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PBeamgroupDirectionCalculator(((nx3.PBeamgroup) (arr.__get(0)) ));
	}
	
	
	public  nx3.PBeamgroup beamgroup;
	
	public   nx3.EDirectionUD getDirection()
	{
		this.topLevel = this.findTopLevel();
		this.bottomLevel = this.findBottomLevel();
		if (( ( this.topLevel + this.bottomLevel ) <= 0 )) 
		{
			return nx3.EDirectionUD.Down;
		}
		
		return nx3.EDirectionUD.Up;
	}
	
	
	public  int topLevel;
	
	public   int findTopLevel()
	{
		int topLevel = ((int) (haxe.lang.Runtime.toInt(this.beamgroup.pnotes.__get(0).nnote.get_topLevel())) );
		if (( this.beamgroup.pnotes.length == 1 )) 
		{
			return topLevel;
		}
		
		{
			int _g1 = 1;
			int _g = this.beamgroup.pnotes.length;
			while (( _g1 < _g ))
			{
				int i = _g1++;
				int level = ((int) (haxe.lang.Runtime.toInt(this.beamgroup.pnotes.__get(i).nnote.get_topLevel())) );
				{
					double x = java.lang.Math.min(((double) (topLevel) ), ((double) (level) ));
					topLevel = ((int) (x) );
				}
				
			}
			
		}
		
		return topLevel;
	}
	
	
	public  int bottomLevel;
	
	public   int findBottomLevel()
	{
		int bottomLevel = ((int) (haxe.lang.Runtime.toInt(this.beamgroup.pnotes.__get(0).nnote.get_bottomLevel())) );
		if (( this.beamgroup.pnotes.length == 1 )) 
		{
			return bottomLevel;
		}
		
		{
			int _g1 = 1;
			int _g = this.beamgroup.pnotes.length;
			while (( _g1 < _g ))
			{
				int i = _g1++;
				int level = ((int) (haxe.lang.Runtime.toInt(this.beamgroup.pnotes.__get(i).nnote.get_bottomLevel())) );
				{
					double x = java.lang.Math.max(((double) (bottomLevel) ), ((double) (level) ));
					bottomLevel = ((int) (x) );
				}
				
			}
			
		}
		
		return bottomLevel;
	}
	
	
	@Override public   double __hx_setField_f(java.lang.String field, double value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef632 = true;
			switch (field.hashCode())
			{
				case -626936807:
				{
					if (field.equals("bottomLevel")) 
					{
						__temp_executeDef632 = false;
						this.bottomLevel = ((int) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -983992529:
				{
					if (field.equals("topLevel")) 
					{
						__temp_executeDef632 = false;
						this.topLevel = ((int) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef632) 
			{
				return super.__hx_setField_f(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef633 = true;
			switch (field.hashCode())
			{
				case -626936807:
				{
					if (field.equals("bottomLevel")) 
					{
						__temp_executeDef633 = false;
						this.bottomLevel = ((int) (haxe.lang.Runtime.toInt(value)) );
						return value;
					}
					
					break;
				}
				
				
				case -1993942992:
				{
					if (field.equals("beamgroup")) 
					{
						__temp_executeDef633 = false;
						this.beamgroup = ((nx3.PBeamgroup) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -983992529:
				{
					if (field.equals("topLevel")) 
					{
						__temp_executeDef633 = false;
						this.topLevel = ((int) (haxe.lang.Runtime.toInt(value)) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef633) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef634 = true;
			switch (field.hashCode())
			{
				case 757123232:
				{
					if (field.equals("findBottomLevel")) 
					{
						__temp_executeDef634 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("findBottomLevel"))) );
					}
					
					break;
				}
				
				
				case -1993942992:
				{
					if (field.equals("beamgroup")) 
					{
						__temp_executeDef634 = false;
						return this.beamgroup;
					}
					
					break;
				}
				
				
				case -626936807:
				{
					if (field.equals("bottomLevel")) 
					{
						__temp_executeDef634 = false;
						return this.bottomLevel;
					}
					
					break;
				}
				
				
				case -972195639:
				{
					if (field.equals("getDirection")) 
					{
						__temp_executeDef634 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getDirection"))) );
					}
					
					break;
				}
				
				
				case -837613560:
				{
					if (field.equals("findTopLevel")) 
					{
						__temp_executeDef634 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("findTopLevel"))) );
					}
					
					break;
				}
				
				
				case -983992529:
				{
					if (field.equals("topLevel")) 
					{
						__temp_executeDef634 = false;
						return this.topLevel;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef634) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   double __hx_getField_f(java.lang.String field, boolean throwErrors, boolean handleProperties)
	{
		{
			boolean __temp_executeDef635 = true;
			switch (field.hashCode())
			{
				case -626936807:
				{
					if (field.equals("bottomLevel")) 
					{
						__temp_executeDef635 = false;
						return ((double) (this.bottomLevel) );
					}
					
					break;
				}
				
				
				case -983992529:
				{
					if (field.equals("topLevel")) 
					{
						__temp_executeDef635 = false;
						return ((double) (this.topLevel) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef635) 
			{
				return super.__hx_getField_f(field, throwErrors, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef636 = true;
			switch (field.hashCode())
			{
				case 757123232:
				{
					if (field.equals("findBottomLevel")) 
					{
						__temp_executeDef636 = false;
						return this.findBottomLevel();
					}
					
					break;
				}
				
				
				case -972195639:
				{
					if (field.equals("getDirection")) 
					{
						__temp_executeDef636 = false;
						return this.getDirection();
					}
					
					break;
				}
				
				
				case -837613560:
				{
					if (field.equals("findTopLevel")) 
					{
						__temp_executeDef636 = false;
						return this.findTopLevel();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef636) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("bottomLevel");
		baseArr.push("topLevel");
		baseArr.push("beamgroup");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


